File: web/src/services/authService.ts
// 重新导出所有认证相关类型定义
export type {
  LoginRequest,
  LoginResponse,
  RegisterRequest,
  RegisterResponse,
  UserInfo
} from './model';

// 重新导出所有认证相关API函数
export {
  login,
  register,
  logout
} from './api';

// 获取当前用户信息
export const getCurrentUser = (): any | null => {
  try {
    const userInfoStr = localStorage.getItem('userInfo');
    if (userInfoStr) {
      return JSON.parse(userInfoStr);
    }
    return null;
  } catch (error) {

    return null;
  }
};

// 检查用户是否已登录
export const isLoggedIn = (): boolean => {
  try {
    const loggedIn = localStorage.getItem('isLoggedIn');
    return loggedIn === 'true';
  } catch (error) {

    return false;
  }
};

// 清除用户登录状态（不调用API）
export const clearUserSession = (): void => {
  localStorage.removeItem('userInfo');
  localStorage.removeItem('isLoggedIn');
  localStorage.removeItem('token');
};

// 保存用户登录状态
export const saveUserSession = (userInfo: any, token: string): void => {
  localStorage.setItem('userInfo', JSON.stringify(userInfo));
  localStorage.setItem('isLoggedIn', 'true');
  localStorage.setItem('token', token);
};

File: src/main/java/com/upec/factoryscheduling/auth/service/AuthService.java
package com.upec.factoryscheduling.auth.service;

import com.upec.factoryscheduling.auth.dto.LoginRequest;
import com.upec.factoryscheduling.auth.dto.LoginResponse;
import com.upec.factoryscheduling.auth.dto.RegisterRequest;
import com.upec.factoryscheduling.auth.entity.User;

/**
 * 认证服务接口
 */
public interface AuthService {

    /**
     * 用户登录
     * @param loginRequest 登录请求
     * @return 登录响应
     */
    LoginResponse login(LoginRequest loginRequest);

    /**
     * 用户注册
     * @param registerRequest 注册请求
     * @return 注册的用户
     */
    User register(RegisterRequest registerRequest);

    /**
     * 用户登出
     */
    void logout();
}
